xbmc/pvr/recordings/PVRRecordings.cpp

```diff
@@ -465,7 +465,12 @@ CPVRRecordingPtr CPVRRecordings::GetRecordingForEpgTag(const EPG::CEpgInfoTagPtr
     else
     {
       // uid is optional, so check other relevant data.
-      if (recording.second->Channel() == epgTag->ChannelTag() &&
```

Comment:
@fritsch Although the diff looks like it at the first glance, the fix is not circumventing a deadlock caused by CPVRRecording's lock by now calling CPVRRecording methods which are not protected by a lock.  CPVRRecording's mutex is not involved in the deadlock.

It's a deadlock caused by CEpg's instance lock and CPVRChannelGroup's lock.

Here are the involved callstacks and a short explanation of the fix idea: http://pastebin.com/9u6D0VdP